test/Microsoft.AspNetCore.Server.KestrelTests/BadHttpRequestTests.cs

```diff
@@ -1,72 +1,99 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using System.Threading.Tasks;
 using Xunit;
 
 namespace Microsoft.AspNetCore.Server.KestrelTests
 {
     public class BadHttpRequestTests
     {
-        [Theory(Skip = "This test fails intermittently and needs to be investigated.")]
-        [InlineData("/ HTTP/1.1\r\n\r\n")]
-        [InlineData(" / HTTP/1.1\r\n\r\n")]
-        [InlineData("  / HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  / HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  /  HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  HTTP/1.1\r\n\r\n")]
+        // Don't send more data than necessary to fail, otherwise the test might throw if
+        // it's still trying to send characters when the server had already shut down the connection
```

Comment:
I did some testing I figured out this is a Windows-only issue. What happens is that `Socket.Shutdown(SocketShutdown.Both)` sends an RST on Windows, essentially killing the connection and preventing the client from reading any data that had been sent previously.

![image](https://cloud.githubusercontent.com/assets/924164/15339667/634f8d72-1c3a-11e6-8f3c-5c8d90f7a8e0.png)


I did the same test on Linux and noticed the same call sends a FIN. tcpdump output:

    08:42:14.193331 IP 127.0.0.1.8080 > 127.0.0.1.40060: Flags [F.], seq 9, ack 8, win 342, options [nop,nop,TS val 902865888 ecr 902865887], length 0